lib/galaxy/schema/fields.py

```diff
@@ -11,18 +12,51 @@
     PlainSerializer,
     WithJsonSchema,
 )
+from pydantic_core import PydanticCustomError
 from typing_extensions import (
     Annotated,
     get_args,
 )
 
+from galaxy.exceptions import MessageException
+
 if TYPE_CHECKING:
     from galaxy.security.idencoding import IdEncodingHelper
 
 ENCODED_DATABASE_ID_PATTERN = re.compile("f?[0-9a-f]+")
 ENCODED_ID_LENGTH_MULTIPLE = 16
 
 
+def validation_message_wrapper(callable: Callable):
+    """Wraps MessageException in a PydanticCustomError."""
+
+    def wrapper(database_id):
+        try:
+            return callable(database_id)
+        except MessageException as e:
+            # If the exception is a MessageException, we want to return it as a PydanticCustomError
+            # so that it can be handled by the Pydantic error handling system, so we can restore the
+            # original exception.
+            if isinstance(e, MessageException):
```

Comment:
I don't think this `if` is needed? If you are inside this exception handler, `e` is surely an instance of `MessageException`.

Reply:
Thanks!